sht3x/sht3x.c

```diff
@@ -92,38 +106,157 @@ int16_t sht3x_read(int32_t* temperature, int32_t* humidity) {
     return SHT_DRV_VERSION_STR;
 }
 
-uint8_t sht3x_get_configured_address(void) {
-    return SHT3X_ADDRESS;
+int16_t sht3x_set_alert_thd(sht3x_i2c_addr_t addr, sht3x_alert_thd_t thd,
+                            uint16_t humidity, int16_t temperature) {
+    int16_t ret;
+    uint32_t tmp;
+    uint16_t limitVal = 0U;
+
+    /* convert inputs to alert threshold word */
+    tmp = humidity * 65535U;
+    tmp = tmp / 1000U; /*humidity was provided in 10%RH*/
+    limitVal = ((uint16_t)tmp & SHT3X_HUMIDITY_LIMIT_MSK);
+
+    tmp = (uint32_t)(temperature + 450);
+    tmp = tmp * 65535U;
+    tmp = tmp / 1750U;
+    tmp = (tmp >> 7);
+    limitVal |= ((uint16_t)tmp & SHT3X_TEMPERATURE_LIMIT_MSK);
+
+    switch (thd) {
+        case SHT3X_HIALRT_SET:
+            ret = sensirion_i2c_write_cmd_with_args(
+                addr, SHT3X_CMD_WRITE_HIALRT_LIM_SET, &limitVal, 1);
+            break;
+
+        case SHT3X_HIALRT_CLR:
+            ret = sensirion_i2c_write_cmd_with_args(
+                addr, SHT3X_CMD_WRITE_HIALRT_LIM_CLR, &limitVal, 1);
+            break;
+
+        case SHT3X_LOALRT_CLR:
+            ret = sensirion_i2c_write_cmd_with_args(
+                addr, SHT3X_CMD_WRITE_LOALRT_LIM_CLR, &limitVal, 1);
+            break;
+
+        case SHT3X_LOALRT_SET:
+            ret = sensirion_i2c_write_cmd_with_args(
+                addr, SHT3X_CMD_WRITE_LOALRT_LIM_SET, &limitVal, 1);
+            break;
+
+        default:
+            ret = STATUS_ERR_INVALID_PARAMS;
+            break;
+    }
+    return ret;
 }
+
+int16_t sht3x_get_alert_thd(sht3x_i2c_addr_t addr, sht3x_alert_thd_t thd,
+                            uint16_t* humidity, int16_t* temperature) {
+
+    int16_t ret;
+    uint16_t word;
+    int32_t tmp;
+
+    switch (thd) {
+        case SHT3X_HIALRT_SET:
+            ret = sensirion_i2c_read_cmd(addr, SHT3X_CMD_READ_HIALRT_LIM_SET,
+                                         &word, 1);
+            break;
+
+        case SHT3X_HIALRT_CLR:
+            ret = sensirion_i2c_read_cmd(addr, SHT3X_CMD_READ_HIALRT_LIM_CLR,
+                                         &word, 1);
+            break;
+
+        case SHT3X_LOALRT_CLR:
+            ret = sensirion_i2c_read_cmd(addr, SHT3X_CMD_READ_LOALRT_LIM_CLR,
+                                         &word, 1);
+            break;
+
+        case SHT3X_LOALRT_SET:
+            ret = sensirion_i2c_read_cmd(addr, SHT3X_CMD_READ_LOALRT_LIM_SET,
+                                         &word, 1);
+            break;
+
+        default:
+            ret = STATUS_ERR_INVALID_PARAMS;
+            break;
+    }
+
+    /* convert threshold word to alert settings in 10*%RH & 10*°C */
+    tmp = (int32_t)(word & SHT3X_HUMIDITY_LIMIT_MSK); /*only 7MSbits*/
+    tmp = (1000 * tmp) / 65535;
+    *humidity = tmp;
+
+    tmp = (int32_t)(word & SHT3X_TEMPERATURE_LIMIT_MSK); /*only 9LSbits*/
+    tmp = (tmp << 7);
+    tmp = (tmp * 1750) / 65535;
+    tmp = tmp - 450;
```

Comment:
Same here. I would prefer to keep everything in the same format (value * 1000). Same as above we can then use the same conversion functions.

```c
uint16_t t_raw, rh_raw;
rh_raw = (word & SHT3X_HUMIDITY_LIMIT_MSK)
t_raw = ((word & SHT3X_TEMPERATURE_LIMIT_MSK) << 7)

tick_to_humidity(rh_raw, &humidity);
tick_to_temperature(t_raw, &temperature);
```

Of course we can then us the same functions for `sht3x_read`